### ISSUE TITLE
review our TLS support

### ISSUE THREAD

**Issue by Gadreel** [Gadreel/divconq#92]
SSL Labs:

```
We made a series of improvements to our grading criteria, to keep up with recent discoveries and continue to nudge site operators to improve their configurations:
Servers that support SSL 3 are now capped at B (C if vulnerable to POODLE).
Grade F given to servers that support only SSL 3.
Servers that support RC4 are capped at B.
Incomplete certificate chains capped at B.
A+ servers are now expected to support TLS_FALLBACK_SCSV and have a SHA2 certificate chain.
The Client Capabilities test has been extended to test client support for all protocol versions, from the first SSL 2 until the most recent TLS 1.2. There have been many improvements to handle various edge cases.
Our SSL/TLS Deployment Best Practices guide has been fully updated.
SSL Labs APIs and our open source client tool for automated bulk testing have become available for early access. We are very excited about our making our APIs publicly available and hope that this will make it easier for system administrators to keep an eye on their systems.
There have been many small improvements throughout, thanks for the attention to detail of the members of our growing SSL Labs community.
``` 

Looks like maybe RC4 is worse than BEAST (CBC vulnerability) so maybe switch Loose from RC4 to CBC preference.

**Comment by Gadreel** [comment]
note that AWS cloud front supports:

ECDHE-RSA-AES128-GCM-SHA256

ECDHE-RSA-AES128-SHA256

ECDHE-RSA-AES128-SHA

ECDHE-RSA-AES256-GCM-SHA384

ECDHE-RSA-AES256-SHA384

ECDHE-RSA-AES256-SHA

AES128-GCM-SHA256

AES256-GCM-SHA384

AES128-SHA256

AES256-SHA

AES128-SHA

RC4-MD5

**Comment by Gadreel** [comment]
get rid of CBC - final word:

https://community.qualys.com/blogs/securitylabs/2015/04/23/ssl-labs-rc4-deprecation-plan

**Comment by Gadreel** [comment]
closed by 87c971b97cceec9b3fdace4e551e2fbf3c0bee56

## before scan

Cipher Suites (sorted by strength; the server has no preference)
TLS_RSA_WITH_AES_128_CBC_SHA (0x2f)	128
TLS_DHE_RSA_WITH_AES_128_CBC_SHA (0x33)   DH 1024 bits (p: 128, g: 128, Ys: 128)   FS   **WEAK**	128
TLS_ECDHE_RSA_WITH_RC4_128_SHA (0xc011)   **WEAK**	128
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (0xc013)   ECDH 571 bits (eq. 15360 bits RSA)   FS	128
TLS_RSA_WITH_AES_128_CBC_SHA256 (0x3c)	128
TLS_DHE_RSA_WITH_AES_128_CBC_SHA256 (0x67)   DH 1024 bits (p: 128, g: 128, Ys: 128)   FS   **WEAK**	128
TLS_RSA_WITH_AES_128_GCM_SHA256 (0x9c)	128
TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 (0x9e)   DH 1024 bits (p: 128, g: 128, Ys: 128)   FS   **WEAK**	128
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 (0xc027)   ECDH 571 bits (eq. 15360 bits RSA)   FS	128
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (0xc02f)   ECDH 571 bits (eq. 15360 bits RSA)   FS	128
TLS_RSA_WITH_AES_256_CBC_SHA256 (0x3d)	256
TLS_DHE_RSA_WITH_AES_256_CBC_SHA256 (0x6b)   DH 1024 bits (p: 128, g: 128, Ys: 128)   FS   **WEAK**	256
TLS_RSA_WITH_AES_256_GCM_SHA384 (0x9d)	256
TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 (0x9f)   DH 1024 bits (p: 128, g: 128, Ys: 128)   FS   **WEAK**	256
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 (0xc028)   ECDH 571 bits (eq. 15360 bits RSA)   FS	256
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (0xc030)   ECDH 571 bits (eq. 15360 bits RSA)   FS	256

### Protocol Details

Secure Renegotiation	Supported
**Secure Client-Initiated Renegotiation	Supported   DoS DANGER (more info)**
Insecure Client-Initiated Renegotiation	No
BEAST attack	Not mitigated server-side (more info)   TLS 1.0: 0x2f
POODLE (SSLv3)	No, SSL 3 not supported (more info)
POODLE (TLS)	No (more info)
**Downgrade attack prevention	No, TLS_FALLBACK_SCSV not supported (more info)**
TLS compression	No
**RC4	Yes   WEAK (more info)**
Heartbeat (extension)	No
Heartbleed (vulnerability)	No (more info)
OpenSSL CCS vuln. (CVE-2014-0224)	No (more info)
**Forward Secrecy	With some browsers (more info)**
Next Protocol Negotiation (NPN)	No
Session resumption (caching)	Yes
Session resumption (tickets)	No
OCSP stapling	No
Strict Transport Security (HSTS)	No
Public Key Pinning (HPKP)	No
Long handshake intolerance	No
TLS extension intolerance	No
TLS version intolerance	No
Incorrect SNI alerts	-
SSL 2 handshake compatibility	No

## after scan

TLS_RSA_WITH_AES_128_CBC_SHA (0x2f)	128
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (0xc013)   ECDH 571 bits (eq. 15360 bits RSA)   FS	128
TLS_RSA_WITH_AES_128_CBC_SHA256 (0x3c)	128
TLS_RSA_WITH_AES_128_GCM_SHA256 (0x9c)	128
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 (0xc027)   ECDH 571 bits (eq. 15360 bits RSA)   FS	128
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (0xc02f)   ECDH 571 bits (eq. 15360 bits RSA)   FS	128
TLS_RSA_WITH_AES_256_CBC_SHA256 (0x3d)	256
TLS_RSA_WITH_AES_256_GCM_SHA384 (0x9d)	256
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 (0xc028)   ECDH 571 bits (eq. 15360 bits RSA)   FS	256
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (0xc030)   ECDH 571 bits (eq. 15360 bits RSA)   FS	256

### Protocol Details

Secure Renegotiation	Supported
**Secure Client-Initiated Renegotiation	Supported   DoS DANGER (more info)**
Insecure Client-Initiated Renegotiation	No
BEAST attack	Not mitigated server-side (more info)   TLS 1.0: 0x2f
POODLE (SSLv3)	No, SSL 3 not supported (more info)
POODLE (TLS)	No (more info)
**Downgrade attack prevention	No, TLS_FALLBACK_SCSV not supported (more info)**
TLS compression	No
RC4	No
Heartbeat (extension)	No
Heartbleed (vulnerability)	No (more info)
OpenSSL CCS vuln. (CVE-2014-0224)	No (more info)
**Forward Secrecy	With some browsers (more info)**
Next Protocol Negotiation (NPN)	No
Session resumption (caching)	Yes
Session resumption (tickets)	No
OCSP stapling	No
Strict Transport Security (HSTS)	No
Public Key Pinning (HPKP)	No
Long handshake intolerance	No
TLS extension intolerance	No
TLS version intolerance	No
Incorrect SNI alerts	-
Uses common DH prime	No
SSL 2 handshake compatibility	No
